### DISCUSSION TITLE
Support Ability Question SA Global + BattleResult

### DISCUSSION THREAD

**Discussion by Laserstrahl** [Albeoris/Memoria#185] [Q&A] [answered]
Hey guys, 

first of all thx for all the great work! I really enoy the mod!
I'm trying to setup an XP booster for the XP gained from battles. Therefore I found this on the website:

BattleResult [code=BonusExp] Level < 40 ? 1.5 * BonusExp : (Level < 80 ? 2.5 * BonusExp : 5 * BonusExp) [/code]

It is also in the AbilityFeatures.txt, however as "SA 4", now I wonder if it will work if I add it as "SA Global" will it work?
Also the documentation says something about creating own files for certain SAs... can't I just edit the AbilityFeatures.txt? What is the benefit of putting the abilities to own files, and where to store them, and do I have to beaware of certain syntax for the files?
I appreciate all the work, but the documentation lacks some information - or I cannot find it.

Putting me in the right direction would be much appreciated!

Regards, 
Laser

**Comment by Tirlititi** [comment, marked as answer]
Hello,

Using "SA Global" will always increase the XP gained from battles indeed. However, as for the supporting ability "Level Up", the XP number displayed will not change (an enemy giving 100 XP per character will show a gain of 100 XP on the battle reward screen but will effectively give more if you add a "SA Global" that boosts XP gain).

You may totally change the existing AbilityFeatures.txt (in your folder "FINAL FANTASY IX\StreamingAssets\Data\Characters\Abilities\") if it's for your own usage. The benefit of making your own AbilityFeatures.txt is to have a "mod folder", that is to say a folder containing all the datas of a mod you would want to share with others.

**Reply by FlameHorizon** [reply]
Is there something which is preventing game from showing effective XP?

**Reply by Tirlititi** [reply]
"Level Up" is designed to be a per-character ability, something that increases the XP of characters individually. So when a single character has Level Up, the game can't display its effective XP gain (otherwise the figure would be wrong for the other characters, or they'd have to somehow display different XP values).
The system used here kept that design choice and don't allow a global XP boost. It still applies that "SA Global" feature to each character individually as it is (and since the formula depends on the character's level, the effective XP is not the same for everyone).
